salt/modules/rest_service.py

```diff
@@ -1,99 +1,119 @@
+
 __func_alias__ = {
-    'list_': 'list'
+    'reload_': 'reload'
 }
 
+# Define the module's virtual name
+__virtualname__ = 'service'
+
 
 def __virtual__():
     '''
-    Only work on RestExampleOS
+    Only work on systems that are a proxy minion
     '''
-    # Enable on these platforms only.
-    enable = set((
-        'RestExampleOS',
-    ))
-    if __grains__['os'] in enable:
+    if __grains__['os'] == 'proxy':
         return __virtualname__
     return False
 
 
-def start(name):
+def get_all():
     '''
-    Start the specified service
+    Return a list of all available services
 
     CLI Example:
 
     .. code-block:: bash
 
-        salt '*' rest_service.start <service name>
+        salt '*' service.get_all
     '''
-    return __opts__['proxyobject'].service_start(name)
+    proxy_fn = 'rest_sample'+ '.service_list'
+    return __opts__['proxymodule'][proxy_fn]()
 
 
-def stop(name):
+def start(name, sig=None):
     '''
-    Stop the specified service
+    Start the specified service on the rest_sample
 
     CLI Example:
 
     .. code-block:: bash
 
-        salt '*' rest_service.stop <service name>
+        salt '*' service.start <service name>
     '''
-    return __opts__['proxyobject'].service_stop(name)
 
+    proxy_fn = 'rest_sample'+ '.service_start'
+    return __opts__['proxymodule'][proxy_fn](name)
 
-def restart(name):
+
+def stop(name, sig=None):
     '''
-    Restart the named service
+    Stop the specified service on the rest_sample
 
     CLI Example:
 
     .. code-block:: bash
 
-        salt '*' rest_service.restart <service name>
+        salt '*' service.stop <service name>
     '''
-
-    return __opts__['proxyobject'].service_restart(name)
+    proxy_fn = 'rest_sample'+ '.service_stop'
+    return __opts__['proxymodule'][proxy_fn](name)
 
 
-def status(name):
+def restart(name, sig=None):
     '''
-    Return the status for a service, returns a bool whether the service is
-    running.
+    Restart the specified service with rest_sample
 
     CLI Example:
 
     .. code-block:: bash
 
-        salt '*' rest_service.status <service name>
+        salt '*' service.restart <service name>
     '''
-    return __opts__['proxyobject'].service_status(name)
+
+    proxy_fn = 'rest_sample'+ '.service_restart'
+    return __opts__['proxymodule'][proxy_fn](name)
 
 
-def list_():
+def status(name, sig=None):
     '''
-    List services.
+    Return the status for a service via rest_sample, returns a bool
+    whether the service is running.
 
     CLI Example:
 
     .. code-block:: bash
 
-        salt '*' rest_service.list <service name>
+        salt '*' service.status <service name>
+    '''
+
+    proxy_fn = 'rest_sample' + '.service_status'
+    resp = __opts__['proxymodule'][proxy_fn](name)
+    if resp['comment'] == 'stopped':
+        return False
+    if resp['comment'] == 'running':
+        return True
+
+
+def running(name, sig=None):
+    '''
+    Return whether this service is running.
+    '''
```

Comment:
Can you add `versionadded` labels to this function and the new `enabled` function below?